### ISSUE TITLE
HttpClient instance sharing / Http Interceptors

### ISSUE THREAD

**Issue by nappynapster** [robisim74/angular-l10n#256]
**I'm submitting a...**
```
[ ] bug report
[ ] feature request
[x] support request
```

**Expected behavior**
The HttpClient instance used for asynchronously retrieving the Translation Files should not be equal to the one of the app that's importing the Translation/Localization-Module. 

**Actual behavior**
Due to the shared instance of the HttpClient, all the attached Interceptors are also applied to the Http-Requests that are loading the translations files.
Depending on the interceptors, the requests will eventually fail

**Steps to reproduce the behavior**
Create an interceptor that modifies the header of all requests.
The calls to request any translation file will hold those modified headers.

StackBlitz: https://stackblitz.com/edit/angular-l10n-8vykjb

unfortunately StackBlitz does not support angular-cli assets, so the request fails but you will still see the modified header.

**Specifications: Angular version, library version, environment, browser**
see in StackBlitz

**Comment by pweyrich** [comment]
The question is: Is it possible to make the Localization- or TranslationModule use a fresh instance of the HttpClient where our app wide interceptors are not attached!?

**Comment by robisim74** [comment]
I don't think it's possible to create different instances of `HttpClient`.

I don't even think it's a problem of this library to worry about the interceptors of the app that hosts it.

I think you should check in your interceptors if, for example, the URL of the request is directed to static files like the translation files, and exclude it from the header change (_a do-nothing noop interceptor_).

**Comment by pweyrich** [comment]
Actually we usually need the interceptors to modify the header also if we request static files. Also, in my opinion, that wouldn't be a good practice for an interceptor to handle such special use cases.

I think it would be beneficial if your `L10nTranslationProvider` uses a new, independent instance of HttpClient which directly requests the translation files without going through the interceptor chain of an app. I could also think of it to be part of the configuration of the translation providers.

Nevertheless, for all those who have a similar issue, we found a workaround to achieve the expected behavior. We've created a [custom translation provider](https://robisim74.github.io/angular-l10n/spec/configuration/#using-a-custom-provider) that passes a new instance of the HttpClient to the `L10nTranslationProvider`.

```typescript
@Injectable()
export class TranslationProvider extends L10nTranslationProvider
{
    constructor(private httpBackend:HttpBackend,
                @Inject(L10N_CONFIG) config:L10nConfigRef,
                caching:Caching)
    {
        super(new HttpClient(httpBackend), config, caching);
    }
}
```
This instance is independent and dispatches the requests directly to the http backend.
